packages/app/src/cli/services/generate.ts

```diff
@@ -90,31 +79,36 @@ async function buildPromptOptions(
   }
 }
 
-function checkLimits(templateSpecifications: TemplateSpecification[], app: AppInterface) {
+function checkLimits(
+  templateSpecifications: TemplateSpecification[],
+  specifications: GenericSpecification[],
+  app: AppInterface,
+) {
   const iterateeFunction = (spec: TemplateSpecification) => {
-    const allValid = spec.types.every((type) => !limitReached(app, type))
+    const allValid = spec.types.every((type) => !limitReached(app, specifications, type))
     return allValid ? 'validTemplateSpecifications' : 'templatesOverlimit'
   }
   return groupBy(templateSpecifications, iterateeFunction)
 }
 
-function limitReached(app: AppInterface, type: GenericSpecification) {
-  if (isFunctionExtension(type)) {
+function limitReached(app: AppInterface, specifications: GenericSpecification[], templateType: TemplateType) {
+  const type = templateType.type
+  if (type === 'function') {
     return app.extensions.function.length >= blocks.functions.defaultRegistrationLimit
```

Comment:
there is a single specification called `function` that includes the global limit, since all the templates have the new type=function, it will pick that up and compare. But yeah, we can do that in a different PR!